api/services/workflow_alias_service.py

```diff
@@ -0,0 +1,118 @@
+import logging
+from collections.abc import Sequence
+from typing import TYPE_CHECKING, Union
+
+from pydantic import BaseModel, Field
+from sqlalchemy import and_, func, select
+from sqlalchemy.orm import Session
+
+from libs.uuid_utils import uuidv7
+
+if TYPE_CHECKING:
+    from sqlalchemy.orm import scoped_session
+
+from models import Workflow, WorkflowNameAlias
+
+
+class WorkflowAliasArgs(BaseModel):
+    app_id: str = Field(..., description="App ID")
+    workflow_id: str = Field(..., description="Workflow ID")
+    name: str = Field(..., description="Alias name", max_length=100)
+    created_by: str | None = Field(default=None, description="User ID who created the alias")
+
+
+logger = logging.getLogger(__name__)
+
+
+class WorkflowAliasService:
+    def create_or_update_alias(
+        self,
+        session: Union[Session, "scoped_session"],
+        request: WorkflowAliasArgs,
+    ) -> WorkflowNameAlias:
+        workflow = session.get(Workflow, request.workflow_id)
+        if not workflow:
+            raise ValueError(f"Workflow {request.workflow_id} not found")
+
+        if workflow.version == Workflow.VERSION_DRAFT:
+            raise ValueError("Cannot create or transfer aliases for draft workflows")
+
+        existing_alias = session.scalar(
+            select(WorkflowNameAlias).where(
+                and_(WorkflowNameAlias.app_id == request.app_id, WorkflowNameAlias.name == request.name)
+            )
+        )
+
+        if existing_alias:
+            old_workflow_id = existing_alias.workflow_id
+            existing_alias.workflow_id = request.workflow_id
+            existing_alias.updated_at = func.current_timestamp()
+
+            existing_alias._is_transferred = True  # type: ignore[reportPrivateUsage]
+            existing_alias._old_workflow_id = old_workflow_id  # type: ignore[reportPrivateUsage]
+            return existing_alias
```

Comment:
If you need to access these attributes outside, don't make them as private. Also, don't use type ignore for them.